plugins/llvm/llvm_binary.hpp

```diff
@@ -235,75 +251,87 @@ OutputIterator read(symbol_iterator begin,
 }
 
 std::vector<symbol> read(const ObjectFile* obj) {
-    int size = utils::distance(obj->begin_symbols(),
-                               obj->end_symbols());
+	std::vector<symbol> symbols;
+	auto symbol_sizes = llvm::object::computeSymbolSizes(*obj);
+
+	for (auto s : symbol_sizes)
+		symbols.push_back(symbol(s.first, s.second));
+	
+	/*
+    int size = std::distance(obj->symbols().begin(),
+							 obj->symbols().end()); 
     std::vector<symbol> symbols;
     symbols.reserve(size);
 
-    read(obj->begin_symbols(),
-         obj->end_symbols(),
+    read(obj->symbols().begin(),
+         obj->symbols().end(),
          std::back_inserter(symbols));
+	*/
     return symbols;
 }
 
-
-
-
 std::vector<symbol> read(const COFFObjectFile * obj) {
-    std::vector<symbol> symbols;
-
-    const pe32_header *pe32;
-    if (error_code err = obj->getPE32Header(pe32))
-        llvm_binary_fail(err);
-
-    for (auto it = obj->begin_symbols(); it != obj->end_symbols(); ++it) {
-        auto sym = obj->getCOFFSymbol(it);
-
-        if (!sym) llvm_binary_fail("not a coff symbol");
-
-        const coff_section *sec = nullptr;
-        if (sym->SectionNumber == COFF::IMAGE_SYM_UNDEFINED)
-            continue;
-
-        if (error_code ec = obj->getSection(sym->SectionNumber, sec))
-            llvm_binary_fail(ec);
-
-        if (!sec) continue;
-
-        uint64_t size = (sec->VirtualAddress + sec->SizeOfRawData) - sym->Value;
-
-        for (auto it = obj->begin_symbols(); it != obj->end_symbols(); ++it) {
-            auto next = obj->getCOFFSymbol(it);
-            if (next->SectionNumber == sym->SectionNumber) {
-                auto new_size = next->Value > sym->Value ?
-                    next->Value - sym->Value : size;
-                size = new_size < size ? new_size : size;
-            }
-        }
-
-        auto addr = sec->VirtualAddress + pe32->ImageBase + sym->Value;
-        symbols.push_back(symbol(*it,addr,size));
-    }
-    return symbols;
+  	std::vector<symbol> symbols;
```

Comment:
it looks like that we don't need this function anymore, if `computeSymbolSizes` can handle it for us. 

Please, try the following: remove all overloads of this function, leaving only the most generic one and compare results on different files for each of three supported containers.